### ISSUE TITLE
Cannot read property 'end' on stream during cleanup

### ISSUE THREAD

**Issue by djfarrelly** [nodeca/probe-image-size#6]
Hi there,
I'm having trouble getting to a bottom of an error that has come up using this library in my application. The package version I'm using is 1.2.1 and the error seems to be happening at this [location in stream.js](https://github.com/nodeca/probe-image-size/blob/d0635406cac8dcce29933a13c9e23235c106ab85/stream.js#L39). I've also only found it to happen in the `gif.js` file. This output is from Bugsnag and it's not catching the image that it's probing so I'm unsure which file is causing this issue. I'm curious if someone may have some insight into what is causing this or how I could potentially catch this error and surface what image is causing it?

Thanks!

```
TypeError Cannot read property 'end' of undefined 
    node_modules/probe-image-size/stream.js:39:20 cleanup
    node_modules/probe-image-size/stream.js:62:7 none
    node_modules/probe-image-size/lib/common.js:25:10 none
    node_modules/probe-image-size/lib/parse_stream/gif.js:22:7 none
    node_modules/stream-parser/index.js:270:20 process
    node_modules/stream-parser/index.js:204:14 none
    node_modules/stream-parser/index.js:294:16 none
    node_modules/stream-parser/index.js:176:3 transform [as _transform]
    node_modules/probe-image-size/node_modules/readable-stream/lib/_stream_transform.js:159:10 Transform._read
    node_modules/probe-image-size/node_modules/readable-stream/lib/_stream_transform.js:147:83 Transform._write
    node_modules/probe-image-size/node_modules/readable-stream/lib/_stream_writable.js:319:64 doWrite
    node_modules/probe-image-size/node_modules/readable-stream/lib/_stream_writable.js:308:5 writeOrBuffer
    node_modules/probe-image-size/node_modules/readable-stream/lib/_stream_writable.js:246:11 Writable.write
    _stream_readable.js:561:20 ondata
    events.js:101:20 emitOne
    events.js:188:7 emit
    _stream_readable.js:387:10 Readable.read
    _stream_readable.js:772:26 flow
    _stream_readable.js:752:3 resume_
    internal/process/next_tick.js:74:11 _combinedTickCallback
    internal/process/next_tick.js:122:9 _tickDomainCallback
```

**Comment by puzrin** [comment]
Seems a dirty typo in condition above

Try to change this line https://github.com/nodeca/probe-image-size/blob/d0635406cac8dcce29933a13c9e23235c106ab85/stream.js#L37 to `if (i !== -1)`

Let me know if it helps or more problems exist.

**Comment by puzrin** [comment]
Should be fixed in 2.0.1

**Comment by djfarrelly** [comment]
Thanks @puzrin! I'll update my app to 2.0.1 today 😄
